File: src/led/json/interpreter/strategies/V0_0_0InterpreterStrategy.java
package led.json.interpreter.strategies;

import com.google.gson.Gson;

import led.ColorRGB;
import led.json.LEDJsonProcedure;

public class V0_0_0InterpreterStrategy implements IInterpreterStrategy {
	@Override
	public ColorRGB[] interpretLine(String _line) {
		return null;
	}
	
	@Override
	public LEDJsonProcedure interpretJson(String _json) {
		return new Gson().fromJson(_json, LEDJsonProcedure.class);
	}
}

File: src/main/java/led/json/interpreter/strategies/V0_0_0InterpreterStrategy.kt
package led.json.interpreter.strategies

import com.github.iamniklas.liocore.led.colorspace.ColorRGB
import com.google.gson.Gson
import led.json.LEDJsonProcedure

class V0_0_0InterpreterStrategy : IInterpreterStrategy {
    override fun interpretLine(_line: String?): Array<ColorRGB> {
        return arrayOf()
    }

    override fun interpretJson(_json: String?): LEDJsonProcedure? {
        return Gson().fromJson(_json, LEDJsonProcedure::class.java)
    }
}